src/clusterfuzz/_internal/datastore/data_handler.py

```diff
@@ -120,20 +120,48 @@ def get_testcase_by_id(testcase_id):
   return testcase
 
 
+@memoize.wrap(memoize.Memcache(MEMCACHE_TTL_IN_SECONDS))
+def _get_builtin_fuzzers():
+  return data_types.Fuzzer.query(ndb_utils.is_true(data_types.Fuzzer.builtin))
+
+
 def find_testcase(project_name,
                   crash_type,
                   crash_state,
                   security_flag,
-                  testcase_to_exclude=None):
+                  testcase_to_exclude=None,
+                  fuzz_target=None):
   """Find an open test case matching certain parameters."""
   # Prepare the query.
-  query = data_types.Testcase.query(
+  query_args = [
       data_types.Testcase.project_name == project_name,
       data_types.Testcase.crash_type == crash_type,
       data_types.Testcase.crash_state == crash_state,
       data_types.Testcase.security_flag == security_flag,
       data_types.Testcase.status == 'Processed',
-      ndb_utils.is_true(data_types.Testcase.open))
+      ndb_utils.is_true(data_types.Testcase.open)
+  ]
+  if fuzz_target and environment.get_value('DEDUP_ONLY_SAME_TARGET'):
```

Comment:
Can we key this on a local_config property (keyed on the issue filing project) instead?

Reply:
Is that different for android than for google 3?

Reply:
No, but we could break it out into its own. Env vars are not as reliable and not available consistently in all environments (e.g. App Engine)

Reply:
I'll land this and try it out with one job and then switch to project (an overloaded term :-() based config.

Reply:
SGTM. It's good to keep these in a consistent place and reduce our reliance on env vars going forward. If we ever need this in App Engine in the future, this could get confusing.

Reply:
Yes, I'll switch to a config before using this env var in more jobs then.